lua/auto-hlsearch.lua

```diff
@@ -1,23 +1,39 @@
+local M = {}
+
 local defaults = {
   remap_keys = { "/", "?", "*", "#", "n", "N" },
 }
 
--- Remap provided keys in order to use :AutoHlsearch command
+-- Remap provided keys in order to use activate() function
 -- while keeping the user's keymap configuration
 local function remap_keys(keys)
   local function set(lhs, keymap)
-    local cmd = ":AutoHlsearch<CR>"
-    if keymap and keymap.rhs then
+    if keymap then
```

Comment:
there was a bug indeed. the remap_keys would override user's keymap if they have lua function there